landoapi/workers/landing_worker.py

```diff
@@ -388,43 +407,34 @@ def run_job(
                 "utf-8"
             )
 
+            temporary_exceptions = {
+                TreeClosed: f"Tree {repo.tree} is closed - retrying later.",
+                TreeApprovalRequired: f"Tree {repo.tree} requires approval - retrying later.",
+                LostPushRace: f"Lost push race when pushing to {repo.push_path}.",
+            }
+
```

Comment:
These error messages should be more tightly coupled to the exceptions themselves, either in the exception class definition or passed in the location they are raised (ie `raise LostPushRace(f"Lost push race when pushing to {repo.push_path}.")`).

That might mean we need to pass the whole `repo` object to `hgrepo.push` to create these error messages, or perhaps these three classes should all subclass some `TemporaryFailureException` that has a `.get_error_message(repo)` function.

Reply:
Changed it around a little, I think this last implementation is a good compromise. We don't need to stick with the original text since all we need is the repo info in the exception, and the exception names are pretty descriptive.